google/cloud/bigquery/external_config.py

```diff
@@ -1003,3 +1003,95 @@ def from_api_repr(cls, resource: dict) -> "ExternalConfig":
         config = cls(resource["sourceFormat"])
         config._properties = copy.deepcopy(resource)
         return config
+
+
+class ExternalCatalogDatasetOptions(object):
+    """Options defining open source compatible datasets living in the BigQuery catalog.
+    Contains metadata of open source database, schema or namespace represented
+    by the current dataset.
+
+    Args:
+        defaultStorageLocationUri: Optional. The storage location URI for all
+            tables in the dataset. Equivalent to hive metastore's database
+            locationUri. Maximum length of 1024 characters. (str)
+        parameters: Optional. A map of key value pairs defining the parameters
+            and properties of the open source schema. Maximum size of 2Mib.
+    """
+
+    def __init__(self, defaultStorageLocationUri: Optional[str] = None, parameters: Optional[dict] = None):
+        self._properties = {}
+        if not isinstance(defaultStorageLocationUri, (str, None)):
```

Comment:
I guess you want to verify that `defaultStorageLocationUri` is of type `str` or is `None`, but trying it in Python it doesn't seem to work:
```
>>> isinstance(None, (str, None))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: isinstance() arg 2 must be a type, a tuple of types, or a union
```

Reply:
Nice. Learned something new today.

Generally `None` should be tested using the `is` or `is not` comparisons. 

Per PEP 008:
```
Comparisons to singletons like None should always be done with
is or is not, never the equality operators.
```

I believe we prolly wanna go with something akin to this:

### OPTION 1.
```
if item is None or isinstance(item, str)
```


### OPTION 2.
Behind the scenes, the type of a `None` object is `NoneType`. Knowing this, it is possible to use a workaround:

```
> mylist = ['alpha', 1, 42.0, None]
> for item in mylist:
    print(f"{item}:", isinstance(item, (str, type(None))))
alpha: True
1: False
42.0: False
None: True
```

But I think OPTION 1 is cleaner and more in line with the PEP.

Thoughts? @Linchin

Reply:
I like option 1 as well! I think it's more readable.